src/map/zone_entities.cpp

```diff
@@ -1103,6 +1103,21 @@ CCharEntity* CZoneEntities::GetCharByID(uint32 id)
     return nullptr;
 }
 
+const NPCEntityList_t CZoneEntities::GetNPCsByName(std::string const& pattern)
+{
+    TracyZoneScoped;
+
+    std::vector<CNpcEntity*> npcs;
+
+    for (const auto& npc : m_npcList) {
+        if (matches(pattern, npc.second->name)) {
+            npcs.emplace_back(dynamic_cast<CNpcEntity*>(npc.second));
```

Comment:
If you're doing a `dynamic_cast` you should be checking the result of it, otherwise you're incurring the cost of checking the inheritance tree and discarding the result.

```cpp
if (auto* PNpc = dynamic_cast<...>(...))
{
    npcs.emplace_back(PNpc);
}
```

Reply:
Done